### ISSUE TITLE
[regression] conan 1.25 breaks linking for module libraries

### ISSUE THREAD

**Issue by DoDoENT** [conan-io/conan#6981]
### Environment Details (include every applicable attribute)
  * Operating System+version: MacOS 10.15.4
  * Compiler+version: Xcode 11.4.1
  * Conan version: 1.25.0
  * Python version: 3.7.7

### Description, analysis and solution proposal

With conan 1.25, `sharedlinkflags` are no longer propagated to CMake module libraries that depend on conan package. This can cause link errors because the linker flags are missing.

The reason for this is in generated `conanbuildinfo.cmake` file. This is the (correct) excerpt from file generated by conan 1.24:

```
set_property(TARGET CONAN_PKG::Package PROPERTY INTERFACE_LINK_LIBRARIES 
${CONAN_PACKAGE_TARGETS_PACKAGE} 
${_CONAN_PKG_LIBS_PACKAGE_DEPENDENCIES} 
${CONAN_SHARED_LINKER_FLAGS_PACKAGE_LIST} 
${CONAN_EXE_LINKER_FLAGS_PACKAGE_LIST} ...
```

and conan 1.25 generates this:

```
set_property(TARGET CONAN_PKG::Package PROPERTY INTERFACE_LINK_LIBRARIES 
${CONAN_PACKAGE_TARGETS_PACKAGE} 
${_CONAN_PKG_LIBS_PACKAGE_DEPENDENCIES} 
$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:${CONAN_SHARED_LINKER_FLAGS_PACKAGE_LIST}> 
$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:${CONAN_EXE_LINKER_FLAGS_PACKAGE_LIST}> ...
```

Now, the problematic part is if module library is linked to `CONAN_PKG::Package`, it won't obtain the `sharedlinkflags` of the `CONAN_PKG::Package` because `$<TARGET_PROPERTY:TYPE>` is `MODULE`, not `EXECUTABLE` nor `SHARED_LIBRARY`.

[CMake documentation](https://cmake.org/cmake/help/v3.17/command/add_library.html) for `MODULE` libraries states:

> `MODULE` libraries are plugins that are not linked into other targets but may be loaded dynamically at runtime using dlopen-like functionality.

Basically, those are [shared libraries to which you cannot link at build time](https://stackoverflow.com/a/43466565/213057).

Therefore, I suggest that the generated code should be:

```
set_property(TARGET CONAN_PKG::Package PROPERTY INTERFACE_LINK_LIBRARIES 
${CONAN_PACKAGE_TARGETS_PACKAGE} 
${_CONAN_PKG_LIBS_PACKAGE_DEPENDENCIES} 
$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:${CONAN_SHARED_LINKER_FLAGS_PACKAGE_LIST}> 
$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:${CONAN_SHARED_LINKER_FLAGS_PACKAGE_LIST}> 
$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:${CONAN_EXE_LINKER_FLAGS_PACKAGE_LIST}> ...
```

i.e. that `sharedlinkerflags` are also given to `MODULE` libraries. Alternatively, you could add `modulelinkerflags` to conan, but I think that would not be a good idea (not only because it would be a breaking change, but also because `module` is something cmake-specific (I don't think any other generator has the same concept)).

In case if you are wondering why would anyone use `MODULE` libraries, they are used for creating [XCTest bundles](https://cmake.org/cmake/help/v3.17/module/FindXCTest.html) for iOS.

[Here is the line](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindXCTest.cmake#L122) from CMake source code which shows how `MODULE` library is used for that purpose.

**Comment by memsharded** [comment]
Many thanks for reporting this. Proposed a fix in https://github.com/conan-io/conan/pull/6983, for 1.25.1

**Comment by DoDoENT** [comment]
Update: the `TYPE` of the `TARGET_PROPERTY` for `MODULE` is `MODULE_LIBRARY`, not `MODULE` (CMake is not very consistent with it).

If I manually fix the generated `conanbuildinfo.cmake` (but using `MODULE_LIBRARY` instead of `MODULE` for generator expression comparison), my build works.

**Comment by memsharded** [comment]
Merged #6983
